test/MatrixClientPeg-test.ts

```diff
@@ -77,6 +83,78 @@ describe("MatrixClientPeg", () => {
         expect(peg.userRegisteredWithinLastHours(24)).toBe(false);
     });
 
+    describe(".start extensions", () => {
+        let testPeg: IMatrixClientPeg;
+
+        beforeEach(() => {
+            // instantiate a MatrixClientPegClass instance, with a new MatrixClient
+            testPeg = new PegClass();
+            fetchMockJest.get("http://example.com/_matrix/client/versions", {});
+        });
+
+        describe("cryptoSetup extension", () => {
+            it("should call default cryptoSetup.getDehydrationKeyCallback", async () => {
+                const mockCryptoSetup = new (class extends CryptoSetupExtensionsBase {
+                    SHOW_ENCRYPTION_SETUP_UI = true;
+                    examineLoginResponse = jest.fn();
+                    persistCredentials = jest.fn();
+                    getSecretStorageKey = jest.fn();
+                    createSecretStorageKey = jest.fn();
+                    catchAccessSecretStorageError = jest.fn();
+                    setupEncryptionNeeded = jest.fn();
+                    getDehydrationKeyCallback = jest.fn().mockReturnValue(null);
+                })() as ProvideCryptoSetupExtensions;
```

Comment:
not sure that the `new (class {} ...)` business adds much here?

```suggestion
                const mockCryptoSetup = {
                    SHOW_ENCRYPTION_SETUP_UI = true;
                    examineLoginResponse = jest.fn();
                    persistCredentials = jest.fn();
                    getSecretStorageKey = jest.fn();
                    createSecretStorageKey = jest.fn();
                    catchAccessSecretStorageError = jest.fn();
                    setupEncryptionNeeded = jest.fn();
                    getDehydrationKeyCallback = jest.fn().mockReturnValue(null);
                } as ProvideCryptoSetupExtensions;
```

(you might need `as any as ProvideCryptoSetupExtensions` to keep TS happy).

Reply:
Also, can't you use `MockModuleWithCryptoSetupExtension` here?

Reply:
Likewise below.